src/base/detail/CollectionImpl.hh

```diff
@@ -111,11 +111,21 @@ struct CollectionStorageValidator
     template<class Size, class OtherSize>
     void operator()(Size, OtherSize)
     {
+        /* No validation needed */
     }
 };
 
 template<>
-struct CollectionStorageValidator<Ownership::value>;
+struct CollectionStorageValidator<Ownership::value>
+{
+    template<class Size, class OtherSize>
+    void operator()(Size dst, OtherSize src)
+    {
+        CELER_VALIDATE(dst == src,
```

Comment:
The message (`too large`) does not seem to match the test (`input == output`)

Reply:
By construction this will only be triggered from integer overflow -- dst and src are supposed to encode the same value with different integer types.

Reply:
I see the usage as being:
```
src/base/Collection.i.hh:    detail::CollectionStorageValidator<W2>()(this->size(),
src/base/Collection.i.hh-                                             other.storage().size());
```
what prevents (beside developer carefullness :) ) the 2 sizes to be of the same type but distinct values?

Reply:
Since this is a "detail" implementation function used by a single class, I think it's OK to worry about only the intended use cases...

Reply:
Nevermind, I missed that the usage is essentially to check the copy-constructor did its job.  So indeed fair enough.